packages/components/src/test/table/UsePresentationTable.test.tsx

```diff
@@ -223,24 +218,113 @@ describe("usePresentationTableWithUnifiedSelection", () => {
     expect(replaceSpy).to.have.been.calledOnceWithExactly("UnifiedSelectionContext", {}, [], 1);
   });
 
-  it("returns an array of selectedRows when keys are passed correctly on selectionChange event", async () => {
+  it("returns an array of selectedRows when keys are passed before the table is loaded", async () => {
     const keys = new KeySet([createTestECInstanceKey()]);
     sinon.stub(Presentation.selection, "getSelection").returns(new KeySet([createTestECInstanceKey()]));
 
-    const { descriptor, item } = getMockData();
-    presentationManagerMock.setup(async (x) => x.getContentDescriptor(moq.It.is((options) => options.keys.size === keys.size))).returns(async () => descriptor);
-    presentationManagerMock
-      .setup(async (x) => x.getContentAndSize(moq.It.is((options) => options.keys.size === keys.size)))
-      .returns(async () => ({ content: new Content(descriptor, [item]), size: 1 }));
+    setupPresentationManager();
+    Presentation.selection.addToSelection("UnifiedSelectionContext", initialProps.imodel, keys, 1);
```

Comment:
Could we avoid the situation of partially stubbing some object? In this situation, you're stubbing `getSelection` to always return something and then calling the real `addToSelection` implementation. This makes `getSelection` and `addToSelection` behave inconsistently, because no matter what you add, you'll always get the same result. This makes it very hard to understand the test... IMO, for this test you shouldn't stub selection manager.